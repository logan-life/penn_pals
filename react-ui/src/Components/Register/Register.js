import React, { useState } from "react";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
const initialData = Object.freeze({
  firstname: "",
  lastname: "",
  username: "",
  password: "",
  security_question:
    "What is your oldest sibling’s birthday month and year? (e.g., January 1900)",
  security_answer: "",
  email: "",
});

export default function Register() {
  const [regData, updateRegdata] = useState(initialData);
  const handleChange = (e) => {
    updateRegdata({
      ...regData,
      [e.target.name]: e.target.value.trim(),
    });
  };

  const handleRegister = (e) => {
    fetch("api/register", {
      method: "POST",
      body: JSON.stringify(regData),
      headers: {
        "Content-Type": "application/json",
      },
    })
      .then((res) => {
        if (res.status === 201) {
          document.getElementById("status").innerHTML =
            "Registered successfully";
        } else {
          const error = new Error(res.error);
          throw error;
        }
      })
      .catch((err) => {
        // console.error(err);
        document.getElementById("status").innerHTML =
          "Error - Try again please!";
      });
  };

  return (
    <div className="Register">
      <h1 id="status">Please Register</h1>
      {/* ControlID sets ID on Form.Control and htmlFor on Form.Label         */}

      <Form className="RegisterForm">
        <Form.Group controlID="fn">
          <Form.Label>First Name</Form.Label>
          <Form.Control
            id = "fn"
            type="text"
            placeholder="Enter first name"
            name="firstname"
            onChange={handleChange}
          />
        </Form.Group>

        <Form.Group controlID="ln">
          <Form.Label>Last Name</Form.Label>
          <Form.Control
            id = "ln"
            type="text"
            placeholder="Enter last name"
            name="lastname"
            onChange={handleChange}
          />
        </Form.Group>

        <Form.Group controlID="email">
          <Form.Label>Email address</Form.Label>
          <Form.Control
            id = "email"
            type="email"
            placeholder="Enter email address"
            name="email"
            onChange={handleChange}
          />
        </Form.Group>

        <Form.Group controlID="un">
          <Form.Label>Username</Form.Label>
          <Form.Control
            id = "un"
            type="text"
            placeholder="Enter username"
            name="username"
            onChange={handleChange}
          />
          <Form.Text className="text-muted">
            TODO username requirements here
          </Form.Text>
        </Form.Group>

        <Form.Group controlID="pwd">
          <Form.Label>Password</Form.Label>
          <Form.Control
            id = "pwd"
            type="password"
            placeholder="Enter password"
            name="password"
            onChange={handleChange}
          />
          <Form.Text className="text-muted">
            TODO password requirements here
          </Form.Text>
        </Form.Group>

        <Form.Group controlID="sq">
          <Form.Label>Secret Question</Form.Label>
          <Form.Control
            as="select"
            name="security_question"
            onChange={handleChange}
          >
            <option value="What is your oldest sibling’s birthday month and year? (e.g., January 1900)">
              What is your oldest sibling’s birthday month and year? (e.g.,
              January 1900)
            </option>
            <option value="What was the name of your first stuffed animal?">
              What was the name of your first stuffed animal?
            </option>
            <option value="What is your maternal grandmother's maiden name?">
              What is your maternal grandmother's maiden name?
            </option>
            <option value="What street did you live on in third grade?">
              What street did you live on in third grade?
            </option>
            <option value="What school did you attend for sixth grade?">
              What school did you attend for sixth grade?
            </option>
          </Form.Control>
        </Form.Group>

        <Form.Group controlID="sa">
          <Form.Label>Secret Answer</Form.Label>
          <Form.Control
            id = "sa"
            type="text"
            placeholder="Secret Answer"
            name="security_answer"
            onChange={handleChange}
          />
        </Form.Group>
        <Button
          onClick={handleRegister}
          type="button"
          variant="primary"
          id="regBtn"
        >
          Submit
        </Button>
      </Form>
    </div>
  );
}
