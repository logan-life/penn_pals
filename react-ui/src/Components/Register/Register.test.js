import React from 'react';
import Register from './Register';
const renderer = require('react-test-renderer');

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

// snapshot testing
test('Register view matches snapshot', () => {
const component = renderer.create(<Register />);
let tree = component.toJSON();
expect(tree).toMatchSnapshot();
});