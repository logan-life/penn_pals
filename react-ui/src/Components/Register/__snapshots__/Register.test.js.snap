// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register view matches snapshot 1`] = `
<div
  className="Register"
>
  <h1
    id="status"
  >
    Please Register
  </h1>
  <form
    className="RegisterForm"
  >
    <div
      className="form-group"
      controlID="fn"
    >
      <label
        className="form-label"
      >
        First Name
      </label>
      <input
        className="form-control"
        id="fn"
        name="firstname"
        onChange={[Function]}
        placeholder="Enter first name"
        type="text"
      />
    </div>
    <div
      className="form-group"
      controlID="ln"
    >
      <label
        className="form-label"
      >
        Last Name
      </label>
      <input
        className="form-control"
        id="ln"
        name="lastname"
        onChange={[Function]}
        placeholder="Enter last name"
        type="text"
      />
    </div>
    <div
      className="form-group"
      controlID="email"
    >
      <label
        className="form-label"
      >
        Email address
      </label>
      <input
        className="form-control"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Enter email address"
        type="email"
      />
    </div>
    <div
      className="form-group"
      controlID="un"
    >
      <label
        className="form-label"
      >
        Username
      </label>
      <input
        className="form-control"
        id="un"
        name="username"
        onChange={[Function]}
        placeholder="Enter username"
        type="text"
      />
      <small
        className="text-muted form-text"
      >
        TODO username requirements here
      </small>
    </div>
    <div
      className="form-group"
      controlID="pwd"
    >
      <label
        className="form-label"
      >
        Password
      </label>
      <input
        className="form-control"
        id="pwd"
        name="password"
        onChange={[Function]}
        placeholder="Enter password"
        type="password"
      />
      <small
        className="text-muted form-text"
      >
        TODO password requirements here
      </small>
    </div>
    <div
      className="form-group"
      controlID="sq"
    >
      <label
        className="form-label"
      >
        Secret Question
      </label>
      <select
        className="form-control"
        name="security_question"
        onChange={[Function]}
      >
        <option
          value="What is your oldest sibling’s birthday month and year? (e.g., January 1900)"
        >
          What is your oldest sibling’s birthday month and year? (e.g., January 1900)
        </option>
        <option
          value="What was the name of your first stuffed animal?"
        >
          What was the name of your first stuffed animal?
        </option>
        <option
          value="What is your maternal grandmother's maiden name?"
        >
          What is your maternal grandmother's maiden name?
        </option>
        <option
          value="What street did you live on in third grade?"
        >
          What street did you live on in third grade?
        </option>
        <option
          value="What school did you attend for sixth grade?"
        >
          What school did you attend for sixth grade?
        </option>
      </select>
    </div>
    <div
      className="form-group"
      controlID="sa"
    >
      <label
        className="form-label"
      >
        Secret Answer
      </label>
      <input
        className="form-control"
        id="sa"
        name="security_answer"
        onChange={[Function]}
        placeholder="Secret Answer"
        type="text"
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      id="regBtn"
      onClick={[Function]}
      type="button"
    >
      Submit
    </button>
  </form>
</div>
`;
